dataservice/api/common/validation.py

```diff
@@ -47,3 +49,53 @@ def enum_validation_generator(_enum):
         ', '.join(list(_enum) + list(COMMON_ENUM)))
 
     return OneOf(extended_enum, error=error_message)
+
+
+class FieldValidator(Str):
+    """
+    Custom Field Validator to take field_name in the validation
+    """
+
+    def _validate(self, value):
+        """Perform validation on ``value``. Raise a :exc:`ValidationError`
+        if validation
+        does not succeed.
+        """
+        errors = []
+        kwargs = {}
+        for validator in self.validators:
+            try:
+                try:
+                    r = validator(value, field=self)
```

Comment:
we can remove this `FieldValidator` class once this PR https://github.com/marshmallow-code/marshmallow/issues/289 for marshmallow is merged. in this we are extending `_validate` function to allow field name to the custom validator function